### ISSUE TITLE
Spring Boot 1.4.3 - ConfigurableEmbeddedServletContainer.setDocumentRoot doesn't work anymore

### ISSUE THREAD

**Issue by jdubois** [spring-projects/spring-boot#7756]
This comes from issue https://github.com/jhipster/generator-jhipster/issues/4756 in JHipster.

We've been using the `setDocumentRoot` method for a long time, and it fails to work when we upgrade to Spring Boot 1.4.3. As this might be linked to a Servlet container upgrade, please note we use Undertow.

**Comment by jdubois** [comment]
Sorry, I'm closing this, it looks like this is our fault -> please ignore this ticket, I will re-open it if needed

**Comment by jdubois** [comment]
No, indeed, the method is not working anymore... I'm doing more tests, but it's quite clear the method is not doing its job anymore.

**Comment by jdubois** [comment]
This is linked to #7752

**Comment by snicoll** [comment]
That issue doesn't tell much as why #7752 might be related - You're referring to a "current target/www hack" but I don't know what it is.

**Comment by jdubois** [comment]
Sorry @snicoll -> the hack is just that we do a `setDocumentRoot("target/www")` for our prod configuration (where we put stuff from `src/main/webapp` after it's minified/uglified).
So it's just that the method isn't working anymore with the new Spring Boot devtools: have a look at my commit https://github.com/jhipster/generator-jhipster/commit/a9e5189bf018f932b9b6834e81877e255586308d and this is working.

**Comment by snicoll** [comment]
>  that the method isn't working anymore

What is the link with devtools?

**Comment by jdubois** [comment]
I have no idea... With the devtools 1.4.3 the method doesn't work, and with the devtools 1.4.2 it works again.
[This comment](https://github.com/spring-projects/spring-boot/issues/7752#issuecomment-269134710) from @felx points to a GitHub commit.

**Comment by snicoll** [comment]
Ok, one more try

> the hack is just that we do a setDocumentRoot("target/www")

I am not asking you what is wrong in our code, I am asking you what code in Jhipster triggers a call to that method. And since you mention devtools, that method must be invoked as part of the refresh or something. To be consistent with your report (since you are making that link) the application works fine on startup and then break once you refresh. If it does not work on startup, I have a hard time understanding why devtools is involved.

Long story short, it would be nice to describe how to reproduce the problem. "doesn't work" is not very helpful. Thanks.

**Comment by jdubois** [comment]
We're doing it [here](https://github.com/jhipster/jhipster-sample-app/blob/v3.12.2/src/main/java/io/github/jhipster/sample/config/WebConfigurer.java#L84) so this is in the `customize` method of the `EmbeddedServletContainerCustomizer`.

Anyway we should close this as a duplicate of #7752

**Comment by philwebb** [comment]
Closing as duplicate of #7752

**Comment by snicoll** [comment]
replying to self: the application does not work fine on startup and it does when devtools is removed from the classpath.
